src/coreclr/src/jit/morph.cpp

```diff
@@ -13076,19 +13092,6 @@ GenTree* Compiler::fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac)
                             {
                                 break;
                             }
-
-#ifdef TARGET_ARM
```

Comment:
that block was under `opts.OptimizationEnabled()`, but it was required for correctness, also it did not need to forbid the optimization, so I moved it up and relaxed the condition, so we mark more `IND` as `GTF_IND_UNALIGNED`, for example, `IND(ADD(LCL_VAR, CNT_INT offset))`.

We still can force jit to generate an instruction that requires floating alignment and get runtime error, for example, with Unsafe API, but it is defiantly not a part of this regression fix, it is another preexisting issue.